src/logic/calculate.js

```diff
@@ -0,0 +1,105 @@
+import operate from './operate'
+const calculate = (data, buttonName) => {
+  let newData = Object.assign({}, data);
+
+  switch (buttonName) {
+    case 'AC':
+    newData = {
+      total: '0',
+      next: null,
+      operation: null,
+    };
+    break;
+
+    case '+/-':
+    if (data.next) {
+      newData.next = (newData.next * -1).toString();
+    } else {
+      newData.total = (newData.total * -1).toString();
+    }
+    break;
+
+    case '.':
+      if (data.operation) {
+        if (data.operation === '=') {
+          newData = {
+          total: '0.',
+          next: null,
+          operation: null,
+        };
+      } else if (newData.next === null) {
+        newData.next = '0.';
+      } else if (!data.next.split('').includes('.')) {
+        newData.next += '.';
+      }
+      } else if (newData.total === null) {
+        newData.total = '0.';
+      } else if (!data.total.split('').includes('.')) {
+        newData.total += '.';
+      }
+      break;
+
+    case '+':
+    case '-':
+    case 'x':
+    case '÷':
+    case '%':
+      if (newData.total === null || newData.total === '0') {
+        newData.total = '0';
+        newData.operation = buttonName;
+      } else if (newData.next === null) {
+        newData.operation = buttonName;
+      } else if (newData.next === '0' && newData.operation === '/') {
+        newData = {
+        total: null,
+        next: null,
+        operation: null,
+        };
+      } else {
+        newData.total = operate(newData.total, newData.next, newData.operation);
+        newData.operation = buttonName;
+        newData.next = null;
+      }
+      break;
+
+    case '=':
+      if (newData.next === '0' || newData.next === '÷') {
+        newData = {
```

Comment:
### Related to division by zero error

What we are doing here is well and good. But insufficient. We should keep a local error state in our `App.js` which will have an error message if the `if newData.next === 0` condition in line 66 is met.